zcash_primitives/src/transaction/builder.rs

```diff
@@ -226,19 +277,24 @@ impl BuildConfig {
     }
 
     /// Returns the Orchard bundle type and anchor for this configuration.
-    pub fn orchard_builder_config(
-        &self,
-    ) -> Option<(orchard::builder::BundleType, orchard::Anchor)> {
+    pub fn orchard_builder_config(&self) -> Option<(BundleType, orchard::Anchor)> {
         match self {
             BuildConfig::Standard { orchard_anchor, .. } => orchard_anchor
                 .as_ref()
-                .map(|a| (orchard::builder::BundleType::DEFAULT_VANILLA, *a)),
-            BuildConfig::Coinbase => Some((
-                orchard::builder::BundleType::Coinbase,
-                orchard::Anchor::empty_tree(),
-            )),
+                .map(|a| (BundleType::DEFAULT_VANILLA, *a)),
+            BuildConfig::Zsa { orchard_anchor, .. } => orchard_anchor
+                .as_ref()
+                .map(|a| (BundleType::DEFAULT_ZSA, *a)),
+            BuildConfig::Coinbase => Some((BundleType::Coinbase, orchard::Anchor::empty_tree())),
         }
     }
+
+    pub fn orchard_bundle_type<FE>(&self) -> Result<BundleType, Error<FE>> {
+        let (bundle_type, _) = self
+            .orchard_builder_config()
+            .ok_or(Error::OrchardBuilderNotAvailable)?;
```

Comment:
It's because of incosistency between Orchard and previous versions - for Orchard there is no more Builder, just a modifiable Bundle, which kind of a Builder but with a different name. I'll rename the error.

Regarding text: I'll change it to the one they have in comments, it is a little more clear